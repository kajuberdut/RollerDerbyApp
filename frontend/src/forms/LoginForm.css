.LoginForm {
    margin-top: 200px;
    margin-left: 12%;    
    width: 43%;
    background-color: #d1d2d4;
    box-shadow: 0 2px 2px 0 rgba(0,0,0,0.14), 0 3px 1px -2px rgba(0,0,0,0.12), 0 1px 5px 0 rgba(0,0,0,0.2);
    /* overflow: hidden; */
    /* overflow-x: hidden; */
}


/* .App .LoginForm {
    overflow: hidden;
  } */