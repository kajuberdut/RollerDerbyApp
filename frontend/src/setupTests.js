// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import '@testing-library/jest-dom';
import './setupTests.js';

// https://stackoverflow.com/questions/32911630/how-do-i-deal-with-localstorage-in-jest-tests
const localStorageMock = {
    getItem: jest.fn(),
    setItem: jest.fn(),
    clear: jest.fn()
  };
  global.localStorage = localStorageMock;