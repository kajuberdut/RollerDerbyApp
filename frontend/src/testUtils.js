import React from "react";
import UserContext from "./multiUse/UserContext";

const demoUser = {
    userId: 1,
    username: "testUser",
    email: "testuser@gmail.com",
    facebookName: null,
    about: null,
    primaryNumber: null,
    level: null,
    ruleset: [],
    position: [],
    insurance: [],
    locationId: null,
    associatedLeagues: null,
    phoneNumber: null,
    firstName: null,
    lastName: null,
    additionalInfo: null,
    secondaryNumber: null
  }

const UserProvider =
    ({ children, user = demoUser }) => (
    <UserContext.Provider value={{ user }}>
      {children}
    </UserContext.Provider>
);



export { UserProvider };
