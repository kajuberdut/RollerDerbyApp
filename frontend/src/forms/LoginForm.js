import React, { useState } from "react";
import { useNavigate } from "react-router-dom";
import "./LoginForm.css";
import {
    Card,
    CardBody,
    CardTitle,
    Form, 
    FormGroup,
    Label,
    Input, 
    Button,
    FormFeedback
  } from "reactstrap";


/** 
 * Login form 
 *
 */

const LoginForm = ({login}) => {
    
   /** Set form data, history, valid, and errorMessage in State */
  let INITIAL_STATE = { username: "", password: ""};
  const [formData, setFormData] = useState(INITIAL_STATE);
  // // const history = useHistory();
  // const [ invalid, setInvalid ] = useState(false);
  // const [errorMessage, setErrorMessage] = useState([]);
  const navigate = useNavigate();


  /** Handle submit by either logging and redirecting in or returning an error message */

  const  handleSubmit = async evt => {
    evt.preventDefault();
    setFormData(INITIAL_STATE);
    let result = await login(formData); 
    
    if(result.success) {
        navigate('/')

    } else {
        let message = result.errors
    //   let message = result.errors[0]
    //   setErrorMessage(message);
    //   setInvalid(true);
  }

  };

  /** Update local state with current state of input element */

  const handleChange = evt => {

    const { name, value }= evt.target;
    setFormData(fData => ({
      ...fData,
      [name]: value,
    }));
    
  };

  /** render form */

  return (
    <section className="col-md-4 LoginForm">
        <Card>
            <CardTitle className="LoginForm-CardTitle">
                <h1>Log In</h1>
            </CardTitle>
            <CardBody>
                <Form onSubmit={handleSubmit}>
                {/* <Form > */}
                  <FormGroup>

                        <Label htmlFor="username" sm={10}>Derby Name: </Label>
                        <Input
                            id="username"
                            name="username"
                            value={formData.username}
                            onChange={handleChange}
                            placeholder="Derby Name"
                            required
                            // invalid={invalid}
                        />

                        <Label htmlFor="password">Password: </Label>
                        <Input
                            type="password"
                            id="password"
                            name="password"
                            value={formData.password}
                            onChange={handleChange}
                            placeholder="Password"
                            required
                            // invalid={invalid}
                        />
{/* 
                    <FormFeedback tooltip>{errorMessage} </FormFeedback> */}

                    </FormGroup>
  
                    <Button>Submit</Button>

                </Form>
            </CardBody>
        </Card>
    </section>
  );
};


export default LoginForm;