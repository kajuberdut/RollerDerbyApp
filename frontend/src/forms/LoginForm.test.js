import { render, screen, cleanup } from '@testing-library/react';
import LoginForm from './LoginForm';
import { UserProvider } from "../testUtils";
import { BrowserRouter } from "react-router-dom";
import '../setupTests.js';

// npm test LoginForm.test.js
// must be in frontend directory 
// ! note that this should be navigating away if the user is present. Not sure what the deal is. 

afterEach(() => {
    cleanup()
  })
  
describe("checks rendering of login form", () => {  

    test('renders login form page', () => {

    render(
        <BrowserRouter>
            <UserProvider>
                <LoginForm />
            </UserProvider>
        </BrowserRouter>);
    });


    test('matches snapshot', function() {
    const { asFragment } = render(
        <BrowserRouter>
            <UserProvider>
                <LoginForm />
            </UserProvider>
        </BrowserRouter>
    );
    expect(asFragment()).toMatchSnapshot(); 
    })

});

describe("asserts login form text", () => {  

    test('renders login form page', () => {

    render(
        <BrowserRouter>
            <UserProvider>
                <LoginForm />
            </UserProvider>
        </BrowserRouter>
    )

        const text = screen.getByText('Login');
        expect(text).toBeInTheDocument();
    });

});