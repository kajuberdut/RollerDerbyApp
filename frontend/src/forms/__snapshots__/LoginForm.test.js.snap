// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checks rendering of login form matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="col-md-4 LoginForm"
  >
    <div
      class="card"
    >
      <div
        class="LoginForm-CardTitle card-title"
      >
        <h1>
          Login
        </h1>
      </div>
      <div
        class="card-body"
      >
        <form
          class=""
        >
          <div
            class="mb-3"
          >
            <label
              class="col-sm-10 col-form-label"
              for="username"
            >
              Derby Name: 
            </label>
            <input
              class="form-control"
              id="username"
              name="username"
              placeholder="Derby Name"
              required=""
              type="text"
              value=""
            />
            <label
              class="form-label"
              for="password"
            >
              Password: 
            </label>
            <input
              class="form-control"
              id="password"
              name="password"
              placeholder="Password"
              required=""
              type="password"
              value=""
            />
          </div>
          <button
            class="btn btn-secondary"
          >
            Submit
          </button>
        </form>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
